crates/plugins/src/manager.rs

```diff
@@ -194,6 +194,48 @@ impl PluginManager {
         };
         Ok(plugin_manifest)
     }
+
+    pub async fn update_lock(&self) -> PluginManagerUpdateLock {
+        let lock = self.update_lock_impl().await;
+        PluginManagerUpdateLock { lock }
+    }
+
+    async fn update_lock_impl(&self) -> anyhow::Result<fd_lock::RwLock<tokio::fs::File>> {
+        let plugins_dir = self.store().get_plugins_directory();
+        tokio::fs::create_dir_all(plugins_dir).await?;
+        let file = tokio::fs::File::create(plugins_dir.join(".updatelock")).await?;
+        let locker = fd_lock::RwLock::new(file);
+        Ok(locker)
+    }
+}
+
+pub struct PluginManagerUpdateLock {
+    lock: anyhow::Result<fd_lock::RwLock<tokio::fs::File>>,
+}
+
+impl PluginManagerUpdateLock {
+    pub fn lock_updates(&mut self) -> PluginManagerUpdateGuard<'_> {
+        if let Ok(lock) = self.lock.as_mut() {
+            let guard = Some(lock.try_write());
+            PluginManagerUpdateGuard { guard }
+        } else {
+            PluginManagerUpdateGuard { guard: None }
+        }
+    }
+}
+
+pub struct PluginManagerUpdateGuard<'lock> {
```

Comment:
"Guards" can get some help from the compiler to make sure they aren't accidentally forgotten:

```suggestion
#[must_use]
pub struct PluginManagerUpdateGuard<'lock> {
```

Reply:
Good call, thanks!